### ISSUE TITLE
Regression: Surface Draw Text when position out of bounds

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#1654]
**Describe the bug**
When text is drawn with position beginning out of the surface area, the text get is drawn different from 3.5.1, with some artifact

**AGS Version**
3.6.0.25

**Game**
[RegressionDrawText.zip](https://github.com/adventuregamestudio/ags/files/8645896/RegressionDrawText.zip)

**To Reproduce**
1. In the Room script, use the following
```AGS Script
// room script file

Overlay* ovr;

function room_AfterFadeIn()
{
  DynamicSprite* dynspr = DynamicSprite.Create(160, 140, true);  
  DrawingSurface* surf = dynspr.GetDrawingSurface();
  surf.DrawingColor = 61765;
  
  surf.DrawString(-10, -5, eFontFont0, "Cut text cut text");
  surf.DrawString(-5, 20, eFontFont0, "Cut text cut text");
  
  surf.Release();
  ovr = Overlay.CreateGraphical(40, 40, 1, true);
}
```
2. Run the game
3. See the error

**Expected behavior**
The text should be drawn normally, but simply cut out correctly.

**Screenshots**
In 3.6.0
![image](https://user-images.githubusercontent.com/2244442/167270860-e2c1cb04-b06a-4953-b6de-5308627a2945.png)

In 3.5.1
![image](https://user-images.githubusercontent.com/2244442/167270865-4766a05a-6a1d-4a12-b7d6-f8dea0773fa1.png)

**Desktop:**
 - OS: Windows
 - Version 10

**Comment by ivan-mogilko** [comment]
Please tell if this happens with WFN (bitmap) fonts or TTF or both?

**Comment by ericoporto** [comment]
I noticed in WFN fonts, undoing this commit fixes: https://github.com/adventuregamestudio/ags/commit/a2f3738f892641bfcd7e7ca930808c52f0757e23 

Now checking if TTF there's anything...

**Comment by ivan-mogilko** [comment]
Yes, setting x = sx and y = sy there is a mistake.

**Comment by ericoporto** [comment]
yeah, the start, I think is h and w, and the length part is correct in ex and ey is correct.

**Comment by ivan-mogilko** [comment]
> Yes, setting x = sx and y = sy there is a mistake.

No, opposite, the `w` and `h` have to be adjusted by the same offset as x and y were.

**Comment by ivan-mogilko** [comment]
Pushed a fix: fdff20fd724a377dc06baf04187e52949d21d923

Also, may propose following script for test:

```
Overlay* ovr;
int draw_x, draw_y;

void Redraw()
{
  DynamicSprite* dynspr = DynamicSprite.Create(120, 20, true);  
  DrawingSurface* surf = dynspr.GetDrawingSurface();
  surf.DrawingColor = 61765;
  
  surf.DrawString(draw_x, draw_y, eFontFont0, "Cut text cut text");
  
  surf.Release();
  ovr = Overlay.CreateGraphical(40, 40, 1, true);
}

function room_AfterFadeIn()
{
  Redraw();
}

function on_key_press(eKeyCode k)
{
  if (k == eKeyLeftArrow)
    draw_x--;
  if (k == eKeyRightArrow)
    draw_x++;
  if (k == eKeyUpArrow)
    draw_y--;
  if (k == eKeyDownArrow)
    draw_y++;
  Redraw();
}
```

**Comment by ericoporto** [comment]
yes, that fixed! Thanks!
